pkg/client/liquidation_recovery.go

```diff
@@ -176,17 +186,59 @@ func handleLiquidationRecovery(
 
 // resolveVbyteFee fetches vByte fee for 25 blocks from the bitcoin handle. If a
 // call to Bitcoin API fails the function catches and logs the error but doesn't
-// fail the execution. If a value of vByte fee was not fetched from the bitcoin
-// handle the function tries to read it from a config file. If the value is not
-// defined in the config file it returns a default vByte fee.
-func resolveVbyteFee(bitcoinHandle bitcoin.Handle, tbtcConfig *tbtc.Config) int32 {
+// fail the execution.
+//
+// If a value of vByte fee was returned from the bitcoin handle it is used to
+// calculate transaction fee estimate. If the estimated transaction fee exceeds
+// the transaction value more than 5%, then the lesser of the suggested vByte fee
+// or configured MaxFeePerVByte is used.
+//
+// If a value of vByte fee was not fetched from the bitcoin handle the function
+// tries to read it from a config file. If the value is not defined in the config file
+// it returns a default vByte fee.
+func resolveVbyteFee(
+	bitcoinHandle bitcoin.Handle,
+	tbtcConfig *tbtc.Config,
+	previousOutputValue int32,
+) int32 {
 	vbyteFee, vbyteFeeError := bitcoinHandle.VbyteFeeFor25Blocks()
 	if vbyteFeeError != nil {
 		logger.Errorf(
 			"failed to retrieve a vbyte fee estimate: [%v]",
 			vbyteFeeError,
 		)
-		// Since the electrs connection is optional, we don't return the error
+		// Since the electrs connection is optional, we don't return the error.
+	}
+	if vbyteFee > 0 {
+		// Fee computation requires that the full transaction be assembled. The
+		// transaction fee should be estimated by assuming the final transaction
+		// will be 175 vBytes, which should be very close to accurate given the
+		// regular structure of liquidation refund transactions.
+		estimatedTransactionFee := vbyteFee * estimatedTransactionSizeVByte
+
+		fivePercentPreviousOutput := previousOutputValue / 20 // 5% of UTXO
+
+		// There is one exception to the rule that the 25-block suggested fee
+		// should be used in the presence of a Bitcoin connection. If this
+		// suggested fee would result in a fee consuming more than 5% of the UTXO
+		// value that is being split, then the lesser of the 25-block suggested
+		// fee and MaxFeePerVByte configured fee should be used.
+		if estimatedTransactionFee > fivePercentPreviousOutput {
+			if tbtcConfig.Bitcoin.MaxFeePerVByte > 0 {
+				vbyteFee = min(vbyteFee, tbtcConfig.Bitcoin.MaxFeePerVByte)
+			}
+
+			// If the final fee is computed as being >5% of UTXO value, the client
+			// should log a WARN message so that the operator can be aware of this
+			// and revise their max fee per vByte if desired.
+			logger.Warnf(
+				"estimated transaction fee [%d] is greater than 5%% of the UTXO value [%d]; "+
```

Comment:
```suggestion
				"estimated transaction fee [%d] is greater than 5% of the UTXO value [%d]; "+
```

Reply:
We need to use a double percent sign (`%%`) for literal value in the format function. 

See https://golang.org/pkg/fmt/#hdr-Printing:
> %%	a literal percent sign; consumes no value

Reply:
!